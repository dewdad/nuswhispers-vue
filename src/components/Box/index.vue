<template>
  <div :class="$style.box">
    <slot />
  </div>
</template>

<style module>
.box {
  background: #fff;
  padding: 1rem;
}

.box p:first-child {
  margin-top: 0;
}

.box p:last-child {
  margin-bottom: 0;
}
</style>
