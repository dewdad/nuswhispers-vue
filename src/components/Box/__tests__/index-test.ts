import Vue from 'vue';
import { createRenderer } from 'vue-server-renderer';
import Box from '../index.vue';

describe('Box', () => {
  const renderer = createRenderer();

  it('renders correctly', () => {
    const vm = new Vue({
      render: (h) => h(Box, 'hello'),
    });
    renderer.renderToString(vm, (err: Error | undefined, html: string) => {
      expect(html).toMatchSnapshot();
    });
  });
});
