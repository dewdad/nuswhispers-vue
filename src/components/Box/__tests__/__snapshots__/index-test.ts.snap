// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `<div data-server-rendered="true" class="box">hello</div>`;
